plugins/Login/Security/LoginFromDifferentCountryDetection.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Login\Security;
+
+use Piwik\Common;
+use Piwik\Container\StaticContainer;
+use Piwik\IP;
+use Piwik\Piwik;
+use Piwik\Plugins\GeoIp2\LocationProvider\GeoIp2;
+use Piwik\Plugins\Login\Emails\LoginFromDifferentCountryEmail;
+use Piwik\Plugins\Login\Model;
+use Piwik\Plugins\UserCountry\LocationProvider;
+use Piwik\Plugins\Login\UserSettings;
+
+class LoginFromDifferentCountryDetection
+{
+    /**
+     * @var Model
+     */
+    private $model;
+
+    /**
+     * @var array|null
+     */
+    private $location = null;
+
+    public function __construct(Model $model)
+    {
+        $this->model = $model;
+    }
+
+    public function isEnabled(): bool
+    {
+        // we need at least one GeoIP provider that is not the default or disabled one
+        $geoIPWorking = $this->isGeoIPWorking();
+
+        // we need the user to have the option enabled - defaults to true if not set
+        $notificationEnabled = $this->isUserNotificationEnabled();
+
+        return $geoIPWorking && $notificationEnabled;
+    }
+
+    private function isGeoIPWorking(): bool
+    {
+        $provider = LocationProvider::getCurrentProvider();
+
+        return $provider instanceof GeoIp2
+            && $provider->isAvailable()
+            && $provider->isWorking()
+            && true === $provider->getSupportedLocationInfo()[LocationProvider::COUNTRY_CODE_KEY];
+    }
+
+    private function isUserNotificationEnabled(): bool
+    {
+        // we need the user to have the option enabled - defaults to true if not set
+        // NOTE: it seems the first read in a new session is always using the default value
+        $userSettings = new UserSettings();
+        return $userSettings->enableLoginCountryChangeNotification->getValue() ?? true;
+
```

Comment:
Let's maybe remove the additional check. It shouldn't happen that `getValue` returns `null`. And if so this would a bug, that shouldn't be hidden that way.
```suggestion
        return $userSettings->enableLoginCountryChangeNotification->getValue();
```